### ISSUE TITLE
"Resource Temporarily Unavailable" error on Windows when removing the temporary directory.

### ISSUE THREAD

**Issue by kingmaker-agm** [ob-ivan/diversitest#25]
The **Cleanup Task** after the completion of the Task (Test Runner) was throwing an **Filesystem Exception** due to the _wrong Directory Separator (slash)_ returned by the `cs278/mktemp` package.

![Error Snapshot](https://storage.googleapis.com/sharing-kingmaker/online-reference/communities/github/ob-vian/diversitest/windows-support-issue/diversitest-failure-message-highlighted.png)

The Source for the Issue is from this particular lines.
https://github.com/ob-ivan/diversitest/blob/30f5e82b7903ccec8d62de7b082d6412018a07c7/src/Command/RunCommand.php#L51

https://github.com/cs278/php-mktemp/blob/9053b3f8f8ec66d7ac2b71605357d7dc195ead8a/src/functions.php#L18-L26

**Comment by kingmaker-agm** [comment]
I will open a **PR** to that package `cs278/mktmp` to make use of the **PHP Constant `DIRECTORY_SEPARATOR`**.

**Comment by kingmaker-agm** [comment]
I have made a **PR** to the `php-mktemp` package.

https://github.com/cs278/php-mktemp/pull/3

Let's wait for them to make the **merge** and release a TAG.

**Comment by ob-ivan** [comment]
@kingmaker-agm Thanks for your involvement! Just a quick question: can you confirm that the problem will be fixed by adjusting the directory separator? For example, if you change the code directly in the `vendor` directory.

**Comment by kingmaker-agm** [comment]
@ob-ivan Still the Error Occurs... I am not sure of the cause of the Error, It just tells "Resource Temporarily Unavailable". Although this **Exception** occurs, the temporary directory contents are deleted.   
  
But when I try to execute the `rmdir` function in a **PHP Interactive Shell**, The Directory is getting removed properly.

**Comment by ob-ivan** [comment]
Then my guess would be that Windows still considers the directory to be "in use" by the time the `rmdir()` call is issued.

This might be due to the fact that the script has changed the working directory to be the temporary directory.

Can you please try out something? Insert these lines:

```php
        $output->writeln('<run>Changing directory back to <run-bold>' . $this->projectRootPath . '</run-bold></run>');
        chdir($this->projectRootPath);
```

right before the lines:

```php
        $output->writeln('<run>Cleaning up temporary directory <run-bold>' . $buildDirectory . '</run-bold></run>');
        $filesystem->remove($buildDirectory);
```

in `\Ob_Ivan\DiversiTest\Command\RunCommand::execute()` and see if the error goes away.

**Comment by kingmaker-agm** [comment]
That **solved** the Error. Thanks @ob-ivan.

**Comment by ob-ivan** [comment]
@kingmaker-agm Thanks for the confirmation!

**Comment by ob-ivan** [comment]
This is released with v0.6.2.
